internal/controller/nonadmindownloadrequest_controller.go

```diff
@@ -0,0 +1,292 @@
+func (r *NonAdminDownloadRequestReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&nacv1alpha1.NonAdminDownloadRequest{}, builder.WithPredicates(ctrlpredicate.Funcs{
+			CreateFunc: func(tce event.TypedCreateEvent[client.Object]) bool {
+				if nadr, ok := tce.Object.(*nacv1alpha1.NonAdminDownloadRequest); ok {
+					return nadr.ReadyForProcessing()
+				}
+				return false
+			},
+			UpdateFunc: func(tue event.TypedUpdateEvent[client.Object]) bool {
+				if nadr, ok := tue.ObjectNew.(*nacv1alpha1.NonAdminDownloadRequest); ok {
+					return nadr.ReadyForProcessing()
+				}
+				return false
+			},
+			DeleteFunc: func(_ event.TypedDeleteEvent[client.Object]) bool {
+				return true
+			}, // we process delete events by deleting corresponding velero download requests if found
+			GenericFunc: func(_ event.TypedGenericEvent[client.Object]) bool {
+				return false
+			},
+		})).
+		Named("nonadmindownloadrequest").
+		Watches(&velerov1.DownloadRequest{}, handler.Funcs{
+			UpdateFunc: func(ctx context.Context, tue event.TypedUpdateEvent[client.Object], rli workqueue.RateLimitingInterface) {
+				if dr, ok := tue.ObjectNew.(*velerov1.DownloadRequest); ok &&
+					dr.Status.Phase == velerov1.DownloadRequestPhaseProcessed { // only reconcile on updates when downloadrequests is processed
+					log := function.GetLogger(ctx, dr, "VeleroDownloadRequestHandler")
+					log.V(1).Info("DownloadRequest populated with url")
+					// on update, we need to reconcile NonAdminDownloadRequests to update
+					rli.Add(reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Namespace: dr.Annotations[constant.NadrOriginNamespaceAnnotation],
+							Name:      dr.Annotations[constant.NadrOriginNameAnnotation],
+						},
+					})
+				}
+			},
+			// DeleteFunc: , TODO: if velero DownloadRequests gets cleaned up, delete this?
+		}, builder.WithPredicates(
+			ctrlpredicate.NewPredicateFuncs(func(object client.Object) bool {
+				// only watch OADP NS
+				if object.GetNamespace() != r.OADPNamespace {
+					return false
+				}
+				// only watch download requests with our label
+				if _, hasUID := object.GetLabels()[constant.NadrOriginNACUUIDLabel]; hasUID {
+					return true
+				}
+				return false
+			}),
+		),
+		).
+		Complete(reconcile.AsReconciler(r.Client, r))
+}
+
+func (r *NonAdminDownloadRequestReconciler) patchAddStatusConditionTypeTrueBackoff(ctx context.Context, req *nacv1alpha1.NonAdminDownloadRequest, typeStr, message string) error {
+	prePatch := req.DeepCopy()
+	req.Status.Phase = nacv1alpha1.NonAdminPhaseBackingOff
+	req.Status.Conditions = []metav1.Condition{
+		{
+			Type:               typeStr,
+			Status:             metav1.ConditionTrue,
+			Reason:             "Error",
+			Message:            message,
+			LastTransitionTime: metav1.Time{Time: time.Now()},
+		},
+	}
+	return r.Status().Patch(ctx, req, client.MergeFrom(prePatch))
```

Comment:
++ I think later we could check where we are using Update() calls and use Patch() on particular parts of status changes. In other parts of the nac controllers for the status we don't use Patch()